packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -126,10 +129,19 @@ class ChildSceneLayer extends Layer {
     this.sceneToken
   });
 
+  /// Offset from parent in the parent's coordinate system.
   Offset offset;
+
+  /// The number of physical pixels the child should produce for each logical pixel.
```

Comment:
Yes, it needs to take that information into account.  The child needs to produce physical pixels so it needs to be informed how much scaling is going to happen by its parents.  (I'm sure we don't compute this accurately yet.)